client.py

```diff
@@ -0,0 +1,307 @@
+                return []
+
+            # Ensure token is in config
+            self._config.api_key['x_authorization'] = self._token.token
+            self._config.api_key_prefix['x_authorization'] = ''
+            _LOGGER.debug("Using token: %s", self._token.token)
+
+            # Reuse existing DefaultApi or create once
+            if self._api is None:
+                self._api = DefaultApi(ApiClient(self._config))
+                _LOGGER.debug("Created new DefaultApi instance")
+    
+            api = self._api
+
+            # Patch logging once
+            if not getattr(api, "_logging_patched", False):
+                original_call_api = api.api_client.call_api
+
+                def logging_call_api(self, resource_path, method, *args, **kwargs):
+                    _LOGGER.debug("Perific API call: %s %s", method, urljoin(self.configuration.host, resource_path))
+                    _LOGGER.debug("call_api args: %s", args)
+                    _LOGGER.debug("call_api kwargs: %s", kwargs)
+                    response = original_call_api(resource_path, method, *args, **kwargs)
+                    try:
+                        _LOGGER.debug("Response status: %s, body: %s", getattr(response, "status", None), getattr(response, "data", None))
+                    except Exception as e:
+                        _LOGGER.debug("Failed logging response: %s", e)
+                    return response
+
+                api.api_client.call_api = logging_call_api.__get__(api.api_client, type(api.api_client))
+                api._logging_patched = True
+
+            try:
+                _LOGGER.debug("Fetching account overview")
+                overview = api.getaccountoverview("application/json")
+                items = getattr(overview, "items", []) or []
+                _LOGGER.debug("Overview items: %s", items)
+                return items
+            except Exception as e:
+                _LOGGER.error("Failed to fetch overview items: %s", e)
+                return []
+
+
+        # Execute in executor since it’s synchronous
+        try:
+            items: list[Item] = await self.hass.async_add_executor_job(_get_overview_items)
+            return [PerificDevice(d) for d in items if isinstance(d, Item)]
+        except Exception as e:
+            _LOGGER.error("Perific get devices unexpected error: %s", e)
+            return []
+  
+    async def async_update_device_metrics(self, device: PerificDevice) -> None:
+        #Fetch latest packets and update metrics for one device.
+        if not self._api:
+            raise RuntimeError("Not authenticated")
+
+        def _get_packets() -> list[LatestItemPackets]:
+            if not self._token or not getattr(self._token, "token", None):
+                _LOGGER.debug("Token missing, quitting...")
+                return []
+
+            # Ensure token is in config (lowercase!)
+            self._config.api_key['x_authorization'] = self._token.token
```

Comment:
can we make this log and auth setup reusable so it won't repeat everywhere? 

Perhaps add a "get_api_client"  method that wires it up?